drivers/mtd/spi_flash.c

```diff
@@ -130,14 +130,18 @@ int spi_mark_badblock(FAR struct spi_flash_dev_s *priv, size_t block)
 
 bool spi_is_badblock(FAR struct spi_flash_dev_s *priv, size_t block)
 {
-  bool ret;
+  bool ret = false;
   off_t address = (block << priv->block_shift);
   uint8_t *spare_buf = kmm_malloc(priv->spare_size);
 
   ssize_t nbytes;
 
+  memset(spare_buf, SPI_ERASED_STATE, priv->spare_size);
```

Comment:
What's the purpose to set spare_buf to all 0xff? If you want to write all 0xff to spare area for bad block, you can do memset before write into bad block spare area. Now it is consuming memory bandwidth no matter you find a bad block or not.